ibm_mq/datadog_checks/ibm_mq/config.py

```diff
@@ -63,9 +63,23 @@ def __init__(self, instance):
         self.channel = instance.get('channel')  # type: str
         self.queue_manager_name = instance.get('queue_manager', 'default')  # type: str
 
-        self.host = instance.get('host', 'localhost')  # type: str
-        self.port = instance.get('port', '1414')  # type: str
-        self.host_and_port = "{}({})".format(self.host, self.port)
+        if not self.channel or not self.queue_manager_name:
+            msg = "channel, queue_manager are required configurations"
+            raise ConfigurationError(msg)
+
+        host = instance.get('host')  # type: str
+        port = instance.get('port')  # type: str
+        self.connection_name = instance.get('connection_name')  # type: str
+        if (host or port) and self.connection_name:
+            raise ConfigurationError(
+                'Provide either host/port or connection_name configurations, not both '
```

Comment:
```suggestion
                'Specify only one host/port or connection_name configuration, '
```
nit: only one might come across more direct imo